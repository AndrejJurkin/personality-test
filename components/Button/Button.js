import React from "react";
import clsx from "clsx";

const Button = React.forwardRef(
  ({ children, primary, size = "medium" }, ref) => {
    return (
      <button
        ref={ref}
        className={clsx("button", {
          ["button--primary"]: primary,
          ["button--small"]: size === "small",
          ["button--medium"]: size === "medium",
          ["button--large"]: size === "large",
        })}
      >
        {children}
      </button>
    );
  }
);

Button.displayName = "Button";

export default Button;
