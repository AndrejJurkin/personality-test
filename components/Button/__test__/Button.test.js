import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../Button";
import React from "react";

describe("Button", () => {
  it("renders button text", () => {
    const { getByText } = render(<Button>Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toBeInTheDocument();
  });

  it("adds correct classes to primary variant", () => {
    const { getByText } = render(<Button primary>Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toHaveClass("button--primary");
  });

  it("adds correct classes to outline variant", () => {
    const { getByText } = render(<Button outline>Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toHaveClass("button--outline");
  });

  it("adds correct classes to disabled variant", () => {
    const { getByText } = render(<Button disabled>Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toHaveClass("button--disabled");
  });

  it("is not disabled by default", () => {
    const { getByText } = render(<Button>Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).not.toHaveClass("button--disabled");
  });

  it("disables button", () => {
    const { getByText } = render(<Button disabled>Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toBeDisabled();
  });

  it("adds correct classes to small variant", () => {
    const { getByText } = render(<Button size="small">Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toHaveClass("button--small");
  });

  it("adds correct classes to medium variant", () => {
    const { getByText } = render(<Button size="medium">Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toHaveClass("button--medium");
  });

  it("adds correct classes to large variant", () => {
    const { getByText } = render(<Button size="large">Click Me</Button>);
    const button = getByText("Click Me");
    expect(button).toHaveClass("button--large");
  });

  it("forwards ref", () => {
    const ref = React.createRef();
    render(<Button ref={ref}>Click Me</Button>);
    expect(ref.current).toBeTruthy();
  });

  it("spreads additional props", () => {
    const { getByText } = render(
      <Button data-testid="test-id">Click Me</Button>
    );
    const button = getByText("Click Me");
    expect(button).toHaveAttribute("data-testid", "test-id");
  });

  it("renders component unchanged", () => {
    const { container } = render(<Button>Test</Button>);
    expect(container).toMatchSnapshot();
  });
});
