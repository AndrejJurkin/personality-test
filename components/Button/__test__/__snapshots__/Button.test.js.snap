// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders component unchanged 1`] = `
<div>
  <button
    class="button button--medium"
  >
    Test
  </button>
</div>
`;
